tests/integration/test_charm.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test):
+    """Build the charm-under-test and deploy it together with related charms.
+
+    Assert on the unit status before any relations/configurations take place.
+    """
+    charm = await ops_test.build_charm(".")
+    resources = {"lego-image": METADATA["resources"]["lego-image"]["upstream-source"]}
+    await ops_test.model.deploy(
+        charm,
+        resources=resources,
+        application_name=APP_NAME,
+        series="jammy",
+        config={
+            "email": "example@email.com",
+            "httpreq_endpoint": "http://dummy.url.com",
+        },
+    )
+
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME],
+        status="active",
+        raise_on_blocked=True,
+        timeout=1000,
+    )
+    assert ops_test.model.applications[APP_NAME].units[0].workload_status == "active"
```

Comment:
not needed for publishing, but probably will be nice to add a test that integrates the charm and verifies that there is an encrypted connection